contracts/libraries/StoryOracleClient.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "./StoryOracleHelper.sol";
+import "../interfaces/IStoryOracleCallback.sol";
+
+
+abstract contract StoryOracleClient {
+    using StoryOracleHelper for StoryOracleHelper.Request;
+
+    uint256 private requestCount = 1;
+    mapping(bytes32 => StoryOracleHelper.Request) private pendingRequests;
+
+    event StoryOracleFulfilled(bytes32 indexed requestId);
+    event StoryOracleCancelled(bytes32 indexed requestId);
+
+    event StoryOracleRequest(
+        bytes32 indexed requestId, // unique id of the oracle request, story oracle backend would call back with id.
+        address requester,  // address of smart contract which sending the oracle request
+        address callbackAddr,  // address of smart contract which implemented the callback function
+        string url,  // url of the json file
```

Comment:
Is it possible to index string type? (URL and path here) Because offchain oracle needs them to fetch data.

Reply:
Yes. added index to url and path.